File: services/api/app/models/lead.py
from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, JSON, String, func

from app.core.database import Base


class Lead(Base):
    __tablename__ = "leads"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    product_type = Column(String, nullable=False)
    lead_type = Column(String, nullable=False)
    status = Column(String, nullable=False, default="new")
    incentive_type = Column(String, nullable=False, default="pending")
    incentive_status = Column(String, nullable=False, default="pending")
    name = Column(String, nullable=False)
    email = Column(String, nullable=True)
    phone = Column(String, nullable=True)
    city = Column(String, nullable=True)
    requirement = Column(String, nullable=True)
    product_details = Column(JSON, nullable=True)
    consent_confirmed = Column(Boolean, nullable=False, default=False)
    convert_to_referral = Column(Boolean, nullable=False, default=False)
    expected_payout = Column(String, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

File: rhserver/platform/models/Lead.js
module.exports = (sequelize, DataTypes) => {
    const Lead = sequelize.define("Lead", {
        product_type: { type: DataTypes.STRING, allowNull: false },
        lead_type: { type: DataTypes.STRING, allowNull: false }, // self, partner, referral, cold

        // Status & Assignment
        status: {
            type: DataTypes.ENUM('New', 'In Progress', 'Credit Approved', 'Disbursed', 'Closed', 'Rejected'),
            defaultValue: "New"
        },
        assignee_id: { type: DataTypes.UUID },
        user_id: { type: DataTypes.INTEGER },

        // Client Details
        name: { type: DataTypes.STRING, allowNull: false },
        email: { type: DataTypes.STRING },
        phone: { type: DataTypes.STRING },
        city: { type: DataTypes.STRING },
        requirement: { type: DataTypes.TEXT },

        product_details: { type: DataTypes.JSONB },

        consent_confirmed: { type: DataTypes.BOOLEAN, defaultValue: false },
        convert_to_referral: { type: DataTypes.BOOLEAN, defaultValue: false }
    }, {
        tableName: 'leads',
        timestamps: true,
        underscored: true
    });

    Lead.associate = function (models) {
        Lead.belongsTo(models.User, { foreignKey: 'user_id', as: 'user' });
        Lead.belongsTo(models.Admin, { foreignKey: 'assignee_id', as: 'assigned_admin' });
    };

    return Lead;
};
